composer/src/composer.rs

```
@@ -932,7 +949,12 @@ impl Builder {
             rust_log_silence: self.rust_log_silence,
         };
 
-        compose.network_id = compose.network_create().await.map_err(|e| e.to_string())?;
+        if self.network_mode.is_none() {
+            compose.network_id = compose.network_create().await.map_err(|e| e.to_string())?;
+        } else {
+            let host_nw = &compose.host_network().await?[0];
+            compose.network_id = host_nw.id.clone().unwrap();
```

Should return an error if no [0] and this unwrap

---

There should always be a host network on the node, so unwrapping panic should be a fair stopper here?

---

I see we already do unwrap on ids else in the code, so it's fine I guess